src/java/htsjdk/samtools/util/Locatable.java

```diff
@@ -22,4 +22,68 @@
      * @return 1-based closed-ended position, undefined if getContig() == null
      */
     int getEnd();
+
+
+    /**
+     * @return number of bases covered by this interval
+     */
+    default int size() {
+        return CoordMath.getLength(getStart(), getEnd());
+    }
+
+    /**
+     * Determines whether this interval overlaps the provided locatable.
+     *
+     * @param other interval to check
+     * @return true if this interval overlaps other, otherwise false
+     */
+    default boolean overlaps(Locatable other) {
+        return withinDistanceOf(other, 0);
+    }
+
+    /**
+     * Determines whether this interval comes within "margin" of overlapping the provided locatable.
+     * This is the same as plain overlaps if margin=0.
+     *
+     * @param other interval to check
+     * @param margin how many bases may be between the two interval for us to still consider them overlapping.
```

Comment:
Also, description seems inaccurate -- if there are 0 bases *between* two intervals, they could be either adjacent or overlapping